vncviewer/Viewport.cxx

```diff
@@ -971,6 +972,12 @@ int Viewport::handleSystemEvent(void *event, void *data)
       keyCode = 0x38;
     }
 
+    // VK_PACKET handling: Translate to WM_*CHAR message, handled below.
+    if (vKey == VK_PACKET) {
+      TranslateMessage(msg);
```

Comment:
I'd rather avoid the risk of intercepting another message type. Was there not enough information in the `WM_KEYDOWN` event? `TranslateMessage()` is able to make sense of it somehow, so there should be something there.

Reply:
I analyzed a hex dump of the entire `WM_KEYDOWN` message, and there is nothing there. `TranslateMessage()` is a Windows API call and need not be limited to the information contained in the parameter. Perhaps there is something in adjacent memory, but even if so it'd be riskier to access that than to use the provided API.

Reply:
I played around with it here and I could not find anything in the event either.

But I was getting proper symbols back from `win32_vkey_to_keysym()`. The complexity from `TranslateMessage()` might not be needed.

Did you test that path?